x-pack/plugins/security_solution/common/endpoint/index_data.ts

```diff
@@ -43,22 +85,80 @@ function delay(ms: number) {
 async function indexHostDocs(
   numDocs: number,
   client: Client,
+  kbnClient: KbnClient & {
+    requestWithApiKey: (path: string, init?: RequestInit | undefined) => Promise<Response>;
+  },
+  realPolicies: Record<string, CreatePackagePolicyResponse['item']>,
+  epmEndpointPackage: GetPackagesResponse['response'][0],
   metadataIndex: string,
-  policyIndex: string,
+  policyResponseIndex: string,
+  enrollFleet: boolean,
   generator: EndpointDocGenerator
 ) {
   const timeBetweenDocs = 6 * 3600 * 1000; // 6 hours between metadata documents
   const timestamp = new Date().getTime();
+  let hostMetadata: HostMetadata;
+  let wasAgentEnrolled = false;
+  let enrolledAgent: undefined | PostAgentEnrollResponse['item'];
+
   for (let j = 0; j < numDocs; j++) {
     generator.updateHostData();
-    generator.updatePolicyId();
+    generator.updateHostPolicyData();
+
+    hostMetadata = generator.generateHostMetadata(timestamp - timeBetweenDocs * (numDocs - j - 1));
+    const { id: appliedPolicyId, name: appliedPolicyName } = hostMetadata.Endpoint.policy.applied;
+
+    // If we don't yet have a "real" policy record, then create it now in ingest (package config)
```

Comment:
The `fleet` CLI flag should drive all of the items created in Fleet (aka: Ingest - they are renaming the entire app to Fleet :) ). so this entire block here wher the policy is created along with the enrolling of a (fake) agent should all be wrapped in the `if (enrollFleet)...`

The main reason (I had) for this is to not impact the teams that works on Resolver - they run this utility (I belive) often, thus we wanted to continue working as it did before.

cc: @oatkiller

Reply:
@marshallmain - FYI ⬆️ 😄